diff --git a/cmake/Findclang.cmake b/cmake/Findclang.cmake
index a95a8c903b..e47d9a0b48 100644
--- a/cmake/Findclang.cmake
+++ b/cmake/Findclang.cmake
@@ -17,10 +17,10 @@ find_path(CLANG_INCLUDE_DIRS NAMES clang/Frontend/ASTUnit.h
 if(${LLVM_LINK_MODE} STREQUAL "shared")
   find_library(CLANG_LIBRARIES
     NAMES
-      libclang-cpp.so.19
-      libclang-cpp.so.19.1
-      clang-cpp-19.0
-      clang-cpp190
+      libclang-cpp.so.20
+      libclang-cpp.so.20.1
+      clang-cpp-20.0
+      clang-cpp200
       clang-cpp
     NAMES_PER_DIR
     HINTS "${LLVM_LIBDIRS}"
diff --git a/cmake/Findlld.cmake b/cmake/Findlld.cmake
index e71838d03d..e6d9ba89ba 100644
--- a/cmake/Findlld.cmake
+++ b/cmake/Findlld.cmake
@@ -9,23 +9,23 @@
 find_path(LLD_INCLUDE_DIRS NAMES lld/Common/Driver.h
     HINTS ${LLVM_INCLUDE_DIRS}
     PATHS
-        /usr/lib/llvm-19/include
-        /usr/local/llvm190/include
-        /usr/local/llvm19/include
-        /usr/local/opt/lld@19/include
-        /opt/homebrew/opt/lld@19/include
-        /home/linuxbrew/.linuxbrew/opt/lld@19/include
+        /usr/lib/llvm-20/include
+        /usr/local/llvm200/include
+        /usr/local/llvm20/include
+        /usr/local/opt/lld@20/include
+        /opt/homebrew/opt/lld@20/include
+        /home/linuxbrew/.linuxbrew/opt/lld@20/include
         /mingw64/include)
 
-find_library(LLD_LIBRARY NAMES lld-19.0 lld190 lld NAMES_PER_DIR
+find_library(LLD_LIBRARY NAMES lld-20.0 lld200 lld NAMES_PER_DIR
     HINTS ${LLVM_LIBDIRS}
     PATHS
-        /usr/lib/llvm-19/lib
-        /usr/local/llvm190/lib
-        /usr/local/llvm19/lib
-        /usr/local/opt/lld@19/lib
-        /opt/homebrew/opt/lld@19/lib
-        /home/linuxbrew/.linuxbrew/opt/lld@19/lib
+        /usr/lib/llvm-20/lib
+        /usr/local/llvm200/lib
+        /usr/local/llvm20/lib
+        /usr/local/opt/lld@20/lib
+        /opt/homebrew/opt/lld@20/lib
+        /home/linuxbrew/.linuxbrew/opt/lld@20/lib
 )
 if(EXISTS ${LLD_LIBRARY})
     set(LLD_LIBRARIES ${LLD_LIBRARY})
@@ -36,12 +36,12 @@ else()
             HINTS ${LLVM_LIBDIRS}
             PATHS
                 ${LLD_LIBDIRS}
-                /usr/lib/llvm-19/lib
-                /usr/local/llvm190/lib
-                /usr/local/llvm19/lib
-                /usr/local/opt/lld@19/lib
-                /opt/homebrew/opt/lld@19/lib
-                /home/linuxbrew/.linuxbrew/opt/lld@19/lib
+                /usr/lib/llvm-20/lib
+                /usr/local/llvm200/lib
+                /usr/local/llvm20/lib
+                /usr/local/opt/lld@20/lib
+                /opt/homebrew/opt/lld@20/lib
+                /home/linuxbrew/.linuxbrew/opt/lld@20/lib
                 /mingw64/lib
                 /c/msys64/mingw64/lib
                 c:/msys64/mingw64/lib)
diff --git a/cmake/Findllvm.cmake b/cmake/Findllvm.cmake
index 5bc874ec42..abdc45c8a0 100644
--- a/cmake/Findllvm.cmake
+++ b/cmake/Findllvm.cmake
@@ -17,12 +17,12 @@ if(ZIG_USE_LLVM_CONFIG)
     # terminate when the right LLVM version is not found.
     unset(LLVM_CONFIG_EXE CACHE)
     find_program(LLVM_CONFIG_EXE
-        NAMES llvm-config-19 llvm-config-19.0 llvm-config190 llvm-config19 llvm-config NAMES_PER_DIR
+        NAMES llvm-config-20 llvm-config-20.0 llvm-config200 llvm-config20 llvm-config NAMES_PER_DIR
         PATHS
             "/mingw64/bin"
             "/c/msys64/mingw64/bin"
             "c:/msys64/mingw64/bin"
-            "C:/Libraries/llvm-19.0.0/bin")
+            "C:/Libraries/llvm-20.0.0/bin")
 
     if ("${LLVM_CONFIG_EXE}" STREQUAL "LLVM_CONFIG_EXE-NOTFOUND")
       if (NOT LLVM_CONFIG_ERROR_MESSAGES STREQUAL "")
@@ -40,9 +40,9 @@ if(ZIG_USE_LLVM_CONFIG)
       OUTPUT_STRIP_TRAILING_WHITESPACE)
 
     get_filename_component(LLVM_CONFIG_DIR "${LLVM_CONFIG_EXE}" DIRECTORY)
-    if("${LLVM_CONFIG_VERSION}" VERSION_LESS 19 OR "${LLVM_CONFIG_VERSION}" VERSION_EQUAL 20 OR "${LLVM_CONFIG_VERSION}" VERSION_GREATER 20)
+    if("${LLVM_CONFIG_VERSION}" VERSION_LESS 20 OR "${LLVM_CONFIG_VERSION}" VERSION_GREATER 20)
       # Save the error message, in case this is the last llvm-config we find
-      list(APPEND LLVM_CONFIG_ERROR_MESSAGES "expected LLVM 19.x but found ${LLVM_CONFIG_VERSION} using ${LLVM_CONFIG_EXE}")
+      list(APPEND LLVM_CONFIG_ERROR_MESSAGES "expected LLVM 20.x but found ${LLVM_CONFIG_VERSION} using ${LLVM_CONFIG_EXE}")
 
       # Ignore this directory and try the search again
       list(APPEND CMAKE_IGNORE_PATH "${LLVM_CONFIG_DIR}")
@@ -66,9 +66,9 @@ if(ZIG_USE_LLVM_CONFIG)
       if (LLVM_CONFIG_ERROR)
         # Save the error message, in case this is the last llvm-config we find
         if (ZIG_SHARED_LLVM)
-          list(APPEND LLVM_CONFIG_ERROR_MESSAGES "LLVM 19.x found at ${LLVM_CONFIG_EXE} does not support linking as a shared library")
+          list(APPEND LLVM_CONFIG_ERROR_MESSAGES "LLVM 20.x found at ${LLVM_CONFIG_EXE} does not support linking as a shared library")
         else()
-          list(APPEND LLVM_CONFIG_ERROR_MESSAGES "LLVM 19.x found at ${LLVM_CONFIG_EXE} does not support linking as a static library")
+          list(APPEND LLVM_CONFIG_ERROR_MESSAGES "LLVM 20.x found at ${LLVM_CONFIG_EXE} does not support linking as a static library")
         endif()
 
         # Ignore this directory and try the search again
